src/resources/assets/js/mixins/largoContainer.vue

```diff
@@ -216,10 +259,47 @@ export default {
             if (videoAnnotations) {
                 annotations = annotations.concat(this.initAnnotations(label, videoAnnotations, VIDEO_ANNOTATION));
             }
+
             // Show the newest annotations (with highest ID) first.
             annotations = annotations.sort((a, b) => b.id - a.id);
 
-            Vue.set(this.annotationsCache, label.id, annotations);
+            return annotations
+        },
+        handleSelectedFilters(filters, union) {
+            union = union ? 1 : 0;
+            this.union = union;
+            if (Object.keys(filters).length > 0) {
+                this.hasActiveFilters = true;
+            } else {
+                this.hasActiveFilters = false;
+            }
+            this.selectedFilters = filters;
+            if (!this.selectedLabel) {
+                return [];
+            }
+            this.loadFilters(this.selectedLabel.id, filters, union);
```

Comment:
If you choose to move the `if` checks out of `loadFilters()` because it works better in `getAnnotations()`, you maybe have to add them here, too. Also you have to add `startLoading` and `finishLoading` here, too.